python/pyspark/ml/clustering.py

```diff
@@ -69,6 +70,25 @@ class KMeans(JavaEstimator, HasFeaturesCol, HasPredictionCol, HasMaxIter, HasTol
     True
     >>> rows[2].prediction == rows[3].prediction
     True
+    >>> import os, tempfile
```

Comment:
I agree with @holdenk . This may be too verbose for a doctest. We can move the temp directory setup in test preparation (where we initialize `sqlContext`) and clean up. We can do that in a separate PR. @holdenk Could you create a JIRA for it? Thanks!